src/ucp/core/ucp_ep.c

```diff
@@ -1016,13 +1016,15 @@ void ucp_ep_destroy(ucp_ep_h ep)
 
 int ucp_ep_config_lane_is_equal(const ucp_ep_config_key_t *key1,
                                 const ucp_ep_config_key_t *key2,
-                                ucp_lane_index_t lane, int compare_types)
+                                ucp_lane_index_t lane1,
+                                ucp_lane_index_t lane2,
+                                int compare_types)
 {
-    return (key1->lanes[lane].rsc_index    == key2->lanes[lane].rsc_index)    &&
-           (key1->lanes[lane].proxy_lane   == key2->lanes[lane].proxy_lane)   &&
-           (key1->lanes[lane].dst_md_index == key2->lanes[lane].dst_md_index) &&
-           (key1->lanes[lane].path_index   == key2->lanes[lane].path_index)   &&
-           ((key1->lanes[lane].lane_types  == key2->lanes[lane].lane_types) ||
+    return (key1->lanes[lane1].rsc_index    == key2->lanes[lane2].rsc_index)    &&
+           (key1->lanes[lane1].proxy_lane   == key2->lanes[lane2].proxy_lane)   &&
+           (key1->lanes[lane1].dst_md_index == key2->lanes[lane2].dst_md_index) &&
+           (key1->lanes[lane1].path_index   == key2->lanes[lane2].path_index)   &&
+           ((key1->lanes[lane1].lane_types  == key2->lanes[lane2].lane_types) ||
```

Comment:
memcmp? or even `(key1->lanes[lane1] ==  key2->lanes[lane2]`

Reply:
no, wouldn't work since we don't want compare lane_tpyes if it wasn't requested

Reply:
sorry, missed extra () around `lane_types` comparison, better don't align different block, maybe even separate by extra line to improve readability?

Reply:
or
```
if (compare_types) {
    return key1->lanes[lane1] == key2->lanes[lane2];
} else {
    return (key1->lanes[lane].rsc_index    == key2->lanes[lane2].rsc_index)    &&
           (key1->lanes[lane].proxy_lane   == key2->lanes[lane2].proxy_lane)   &&
           (key1->lanes[lane].dst_md_index == key2->lanes[lane2].dst_md_index) &&
           (key1->lanes[lane].path_index   == key2->lanes[lane2].path_index);
}
```

Reply:
fixed